pkg/costmodel/allocation_helpers.go

```diff
@@ -614,12 +621,13 @@ func applyRAMBytesUsedMax(podMap map[podKey]*pod, resRAMBytesUsedMax []*prom.Que
 	}
 }
 
-func applyGPUUsageAvg(podMap map[podKey]*pod, resGPUUsageAvg []*prom.QueryResult, podUIDKeyMap map[podKey][]podKey) {
+// same func is used for both GPUUsageAvg and GPUUsageMax
+func applyGPUUsage(podMap map[podKey]*pod, resGPUUsageAvgOrMax []*prom.QueryResult, podUIDKeyMap map[podKey][]podKey, mode string) {
```

Comment:
Could this instead be broken up into 4 functions, for better clarity? It currently handles results from Prometheus queries `resGPUsUsageAvg`, `resGPUsUsageMax`, `resIsGpuShared`, `resGetGPUInfo`.